File: java/src/main/java/com/rillet/codingchallenge/accounting/infra/Configurations.java
package com.rillet.codingchallenge.accounting.infra;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.zalando.jackson.datatype.money.MoneyModule;

@Configuration
public class Configurations {
    @Bean
    public ObjectMapper objectMapper() {
        return new ObjectMapper()
                .registerModule(new JavaTimeModule())
                .registerModule(new MoneyModule().withAmountFieldName("value").withQuotedDecimalNumbers());
    }
}

File: kotlin/src/main/kotlin/com/rillet/codingchallenge/accounting/infra/Configurations.kt
package com.rillet.codingchallenge.accounting.infra

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.zalando.jackson.datatype.money.MoneyModule

@Configuration
class Configurations {
  @Bean
  fun objectMapper(): ObjectMapper =
      jacksonObjectMapper().apply {
        registerModule(JavaTimeModule())
        registerModule(MoneyModule().withAmountFieldName("value").withQuotedDecimalNumbers())
      }
}
